### DISCUSSION TITLE
Question about Running Nhost locally

### DISCUSSION THREAD

**Discussion by PtradeLLC** [nhost/nhost#2356] [Q&A] [answered]
Hello guys,
So we installed Nhost locally on our Next.js app, and configured the `_app.js` to wrap with `NhostClient` with these settings

```
const nhost = new NhostClient({
  subdomain: process.env.NEXT_PUBLIC_NHOST_SUBDOMAIN || ' ',
  region: process.env.NEXT_PUBLIC_NHOST_REGION || ' '
});
``` 
- The subdomain and region are taken from our account via `https://app.nhost.io/`. 

So I guess my question is what makes this installation local one, also can we use the `Run` feature locally or do we have to upgrade to use it, lastly - can we just connect our database directly using `Postgres` to Nhost without using Hasura or having to set it up via `Run`?

I was under the impression that after all is configured and setup we'd be able to go to local port `http://localhost:3001` or something to login to the local Nhost dashboard instead of `https://app.nhost.io/account-name`. Please advice.

Also, how can we make this work (pls see attached img) - We've been having a `Could not verify JWT: JWTExpired` every time.
<img width="1154" alt="Screenshot 2023-11-03 at 2 04 07 PM" src="https://github.com/nhost/nhost/assets/138940763/517a1c14-c4ec-4215-ae71-60ef5ce517dd">

Thanks in advance.

**Comment by dbarrosop** [comment, marked as answer]
> So I guess my question is what makes this installation local one

I am not sure I understand what you mean but if you want to connect to your local environment, when you start the cli you get that information:

```
Nhost development environment started.

URLs:
- Postgres:             postgres://postgres:postgres@localhost:5432/local
- Hasura:               https://local.hasura.nhost.run
- GraphQL:              https://local.graphql.nhost.run
- Auth:                 https://local.auth.nhost.run
- Storage:              https://local.storage.nhost.run
- Functions:            https://local.functions.nhost.run
- Dashboard:            https://local.dashboard.nhost.run
- Mailhog:              https://local.mailhog.nhost.run

SDK Configuration:
 Subdomain:             local         <----------------
 Region:                (empty)       <----------------

Run `nhost up` to reload the development environment
Run `nhost down` to stop the development environment
Run `nhost logs` to watch the logs
```

> can we use the Run feature locally

as of today that's a cloud feature.

> can we just connect our database directly using Postgres to Nhost without using Hasura

yes, you can. You have the connection string in the same CLI info that is presented on screen and in the cloud you can head to settings -> database to see yours.

> We've been having a Could not verify JWT: JWTExpired every time

That's an issue on our dashboard that pops up when the tab is left open in the background for a while. We should be fixing that issue soon.

**Reply by PtradeLLC** [reply]
@dbarrosop -  Wow, Thank you for the wealth of info - very much appreciated. One last issue before I stop bugging you. I just realized that our auth flow is throwing an error:
```
{
    "status": 500,
    "message": "{\"message\":\"field 'authProviderRequest' not found in type: 'query_root': {\\\"response\\\":{\\\"errors\\\":[{\\\"extensions\\\":{\\\"code\\\":\\\"validation-failed\\\",\\\"path\\\":\\\"$.selectionSet.authProviderRequest\\\"},\\\"message\\\":\\\"field 'authProviderRequest' not found in type: 'query_root'\\\"}],\\\"status\\\":200,\\\"headers\\\":{}},\\\"request\\\":{\\\"query\\\":\\\"query providerRequest($id: uuid!) {\\\\n  authProviderRequest(id: $id) {\\\\n    id\\\\n    options\\\\n  }\\\\n}\\\",\\\"variables\\\":{\\\"id\\\":\\\"5260e04e-3241-45b8-ab47-74051c12d338\\\"}}}\",\"stack\":\"Error: field 'authProviderRequest' not found in type: 'query_root': {\\\"response\\\":{\\\"errors\\\":[{\\\"extensions\\\":{\\\"code\\\":\\\"validation-failed\\\",\\\"path\\\":\\\"$.selectionSet.authProviderRequest\\\"},\\\"message\\\":\\\"field 'authProviderRequest' not found in type: 'query_root'\\\"}],\\\"status\\\":200,\\\"headers\\\":{}},\\\"request\\\":{\\\"query\\\":\\\"query providerRequest($id: uuid!) {\\\\n  authProviderRequest(id: $id) {\\\\n    id\\\\n    options\\\\n  }\\\\n}\\\",\\\"variables\\\":{\\\"id\\\":\\\"5260e04e-3241-45b8-ab47-74051c12d338\\\"}}}\\n    at /app/node_modules/.pnpm/graphql-request@3.7.0_graphql@16.2.0/node_modules/graphql-request/dist/index.js:340:31\\n    at step (/app/node_modules/.pnpm/graphql-request@3.7.0_graphql@16.2.0/node_modules/graphql-request/dist/index.js:63:23)\\n    at Object.next (/app/node_modules/.pnpm/graphql-request@3.7.0_graphql@16.2.0/node_modules/graphql-request/dist/index.js:44:53)\\n    at fulfilled (/app/node_modules/.pnpm/graphql-request@3.7.0_graphql@16.2.0/node_modules/graphql-request/dist/index.js:35:58)\\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)\"}",
    "error": "internal-error"
}```

What do you make of this?

**Reply by dbarrosop** [reply]
that looks like either a metadata inconsistency or a permissions issue.

**Reply by PtradeLLC** [reply]
I see. Thanks for the feedback. I noticed that the `graphql` tab is displaying `field 'users' not found in type: 'query_root'` - It's a red flag. However, our team is invested in `prisma` as the ORM to communicate with the server. Do you have any docs / references to using `prisma`?

Entire error:

`{"message":"field 'users' not found in type: 'query_root': {\"response\":{\"errors\":[{\"extensions\":{\"code\":\"validation-failed\",\"path\":\"$.selectionSet.users\"},\"message\":\"field 'users' not found in type: 'query_root'\"}],\"status\":200,\"headers\":{}},\"request\":{\"query\":\"query users($where: users_bool_exp!) {\\n users(where: $where) {\\n ...userFields\\n }\\n}\\n\\nfragment userFields on users {\\n id\\n createdAt\\n disabled\\n displayName\\n avatarUrl\\n email\\n passwordHash\\n emailVerified\\n phoneNumber\\n phoneNumberVerified\\n defaultRole\\n isAnonymous\\n ticket\\n otpHash\\n totpSecret\\n activeMfaType\\n newEmail\\n locale\\n metadata\\n roles {\\n role\\n }\\n}\",\"variables\":{\"where\":{\"email\":{\"_eq\":\"[support@forgedmart.com](mailto:support@forgedmart.com)\"}}}}}","stack":"Error: field 'users' not found in type: 'query_root': {\"response\":{\"errors\":[{\"extensions\":{\"code\":\"validation-failed\",\"path\":\"$.selectionSet.users\"},\"message\":\"field 'users' not found in type: 'query_root'\"}],\"status\":200,\"headers\":{}},\"request\":{\"query\":\"query users($where: users_bool_exp!) {\\n users(where: $where) {\\n ...userFields\\n }\\n}\\n\\nfragment userFields on users {\\n id\\n createdAt\\n disabled\\n displayName\\n avatarUrl\\n email\\n passwordHash\\n emailVerified\\n phoneNumber\\n phoneNumberVerified\\n defaultRole\\n isAnonymous\\n ticket\\n otpHash\\n totpSecret\\n activeMfaType\\n newEmail\\n locale\\n metadata\\n roles {\\n role\\n }\\n}\",\"variables\":{\"where\":{\"email\":{\"_eq\":\"[support@forgedmart.com](mailto:support@forgedmart.com)\"}}}}}\n at /app/node_modules/.pnpm/graphql-request@[3.7.0_graphql@16.2.0](http://mailto:3.7.0_graphql@16.2.0.0/)/node_modules/graphql-request/dist/index.js:340:31\n at step (/app/node_modules/.pnpm/graphql-request@[3.7.0_graphql@16.2.0](http://mailto:3.7.0_graphql@16.2.0.0/)/node_modules/graphql-request/dist/index.js:63:23)\n at Object.next (/app/node_modules/.pnpm/graphql-request@[3.7.0_graphql@16.2.0](http://mailto:3.7.0_graphql@16.2.0.0/)/node_modules/graphql-request/dist/index.js:44:53)\n at fulfilled (/app/node_modules/.pnpm/graphql-request@[3.7.0_graphql@16.2.0](http://mailto:3.7.0_graphql@16.2.0.0/)/node_modules/graphql-request/dist/index.js:35:58)\n at process.processTicksAndRejections (node:internal/process/task_queues:95:5)"}`

`{
"status": 500,
"message": "{\"message\":\"field 'authProviderRequest' not found in type: 'query_root': {\\\"response\\\":{\\\"errors\\\":[{\\\"extensions\\\":{\\\"code\\\":\\\"validation-failed\\\",\\\"path\\\":\\\"$.selectionSet.authProviderRequest\\\"},\\\"message\\\":\\\"field 'authProviderRequest' not found in type: 'query_root'\\\"}],\\\"status\\\":200,\\\"headers\\\":{}},\\\"request\\\":{\\\"query\\\":\\\"query providerRequest($id: uuid!) {\\\\n authProviderRequest(id: $id) {\\\\n id\\\\n options\\\\n }\\\\n}\\\",\\\"variables\\\":{\\\"id\\\":\\\"5260e04e-3241-45b8-ab47-74051c12d338\\\"}}}\",\"stack\":\"Error: field 'authProviderRequest' not found in type: 'query_root': {\\\"response\\\":{\\\"errors\\\":[{\\\"extensions\\\":{\\\"code\\\":\\\"validation-failed\\\",\\\"path\\\":\\\"$.selectionSet.authProviderRequest\\\"},\\\"message\\\":\\\"field 'authProviderRequest' not found in type: 'query_root'\\\"}],\\\"status\\\":200,\\\"headers\\\":{}},\\\"request\\\":{\\\"query\\\":\\\"query providerRequest($id: uuid!) {\\\\n authProviderRequest(id: $id) {\\\\n id\\\\n options\\\\n }\\\\n}\\\",\\\"variables\\\":{\\\"id\\\":\\\"5260e04e-3241-45b8-ab47-74051c12d338\\\"}}}\\n at /app/node_modules/.pnpm/graphql-request@[3.7.0_graphql@16.2.0](http://mailto:3.7.0_graphql@16.2.0.0/)/node_modules/graphql-request/dist/index.js:340:31\\n at step (/app/node_modules/.pnpm/graphql-request@[3.7.0_graphql@16.2.0](http://mailto:3.7.0_graphql@16.2.0.0/)/node_modules/graphql-request/dist/index.js:63:23)\\n at Object.next (/app/node_modules/.pnpm/graphql-request@[3.7.0_graphql@16.2.0](http://mailto:3.7.0_graphql@16.2.0.0/)/node_modules/graphql-request/dist/index.js:44:53)\\n at fulfilled (/app/node_modules/.pnpm/graphql-request@[3.7.0_graphql@16.2.0](http://mailto:3.7.0_graphql@16.2.0.0/)/node_modules/graphql-request/dist/index.js:35:58)\\n at process.processTicksAndRejections (node:internal/process/task_queues:95:5)\"}",
"error": "internal-error"
}`
